metricflow/protocols/sql_client.py

```diff
@@ -197,7 +197,10 @@ class SqlEngineAttributes(Protocol):
     timestamp_type_supported: ClassVar[bool]
     timestamp_to_string_comparison_supported: ClassVar[bool]
     cancel_submitted_queries_supported: ClassVar[bool]
-    percentile_aggregation_supported: ClassVar[bool]
+    continuous_percentile_aggregation_supported: ClassVar[bool]
+    discrete_percentile_aggregation_supported: ClassVar[bool]
+    approximate_continuous_percentile_aggregation_supported: ClassVar[bool]
+    approximate_discrete_percentile_aggregation_supported: ClassVar[bool]
```

Comment:
Ooof. I have a low key goal to eliminate these things altogether, but this extends the set immensely. For now I think it's ok, there isn't really a better way to do this with the way things are structured at the moment, but we need to rethink our approach to this before the divergence grows any further.